addon-test-support/@ember/test-helpers/dom/fire-event.js

```diff
@@ -84,27 +84,39 @@ function buildBasicEvent(type, options = {}) {
 // eslint-disable-next-line require-jsdoc
 function buildMouseEvent(type, options = {}) {
   let event;
+  let eventOpts = merge(merge({}, DEFAULT_EVENT_OPTIONS), options);
   try {
-    event = document.createEvent('MouseEvents');
-    let eventOpts = merge(merge({}, DEFAULT_EVENT_OPTIONS), options);
-    event.initMouseEvent(
-      type,
-      eventOpts.bubbles,
-      eventOpts.cancelable,
-      window,
-      eventOpts.detail,
-      eventOpts.screenX,
-      eventOpts.screenY,
-      eventOpts.clientX,
-      eventOpts.clientY,
-      eventOpts.ctrlKey,
-      eventOpts.altKey,
-      eventOpts.shiftKey,
-      eventOpts.metaKey,
-      eventOpts.button,
-      eventOpts.relatedTarget
-    );
+    event = new MouseEvent(type, eventOpts);
   } catch (e) {
+    // left intentionally blank
```

Comment:
Hmm, aren’t some errors valid here? E.g. when `MouseEvent` is present but you pass invalid options or something?

Reply:
maybe we could make a constant in module scope to track if `MouseEvent‘ is functional, then remove the try/catch inline?

Reply:
Was following the pattern from buildKeyboardEvent, but I agree that the solution checking if MouseEvent exists is better